plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/TemplateListPage.java

```diff
@@ -151,9 +154,15 @@ public void widgetSelected(SelectionEvent e) {
 						try {
 							model.setTemplateFileName(file);
 							return;
-						}catch(ResourceFactoryException | ClassCastException err) {
+						}catch(ClassCastException err) {
```

Comment:
Problem is that this dialog is showing up on 1 to 2 layers away from the error log and the message is just basically saying "whoops - something went wrong and i can't be bothered to give you a hint of what actually went wrong so please now go close all dialogs and lookup the error view which might not even be installed to look for what happened" :)

It is a common good practice in eclipse that:

A) error messages give at least some context
B) have a Details button where underlying errors can be accessed without use of error log view

It thus should be something like:

`ErrorDialog.openError(shell,
                                                MSG_UPLOAD_TEMPLATE,
                                                NLS.bind("Unable to read and/or parse the file \"{0}\" as a template. Erorr: '{1}', err.getMessage(), ValidationStatus.error(e.getMessage(),e));`

I believe that is what is used most places anyway.